app/models/evss_claim_document.rb

```diff
@@ -47,9 +54,41 @@ def ==(other)
     attributes == other.attributes
   end
 
+  def to_serializable_hash
+    # file_obj is not suitable for serialization
+    to_hash.tap { |h| h.delete :file_obj }
+  end
+
   private
 
   def known_document_type?
     errors.add(:base, 'Must use a known document type') unless description
   end
+
+  def unencrypted_pdf?
+    return unless file_name =~ /\.pdf$/i
+    xref = PDF::Reader::XRef.new file_obj.tempfile
+    errors.add(:file_obj, 'PDF must not be encrypted') if xref.trailer[:Encrypt].present?
+    file_obj.tempfile.rewind
+  rescue PDF::Reader::MalformedPDFError
+    errors.add(:file_obj, 'PDF is malformed')
+  end
+
+  def normalize_text
+    return unless file_name =~ /\.txt$/i
+    text = file_obj.read
+    file_obj.rewind
+    cd = CharDet.detect text
+    if cd['confidence'] < 0.5
+      errors.add(:file_obj, 'Cannot guess encoding of text')
+      return false
+    end
+    return if cd['encoding'] == 'ascii'
```

Comment:
There are a few magic strings/numbers here, can we make some of them constants (thinking this and the confidence level)